sdk/keyvault/Azure.Security.KeyVault.Shared/tests/PartiallyDeployedAttribute.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Azure.Security.KeyVault.Tests
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    public class PartiallyDeployedAttribute : NUnitAttribute, IWrapTestMethod
+    {
+        public TestCommand Wrap(TestCommand command) => new PartiallyDeployedCommand(command);
+
+        private class PartiallyDeployedCommand : DelegatingTestCommand
+        {
+            public PartiallyDeployedCommand(TestCommand innerCommand) : base(innerCommand)
+            {
+            }
+
+            public override TestResult Execute(TestExecutionContext context)
+            {
+                context.CurrentResult = innerCommand.Execute(context);
+
+                if (context.CurrentResult.ResultState.Status == TestStatus.Failed &&
+                    context.CurrentResult.Message.Contains("Status: 400") &&
+                    context.CurrentResult.Message.Contains("ErrorCode: BadParameter"))
```

Comment:
Nice use of the wrapper attribute :) - It would be nice if the message text was more specific to the service endpoint not being available. Just out of curiosity, what does the auth challenge return in this scenario?

Reply:
The endpoint is available, but it's a new property on a model that isn't supported. That's been more often the scenario in which this will be handy. I think the reason this started popping up is because I removed a custom assertion I had that basically did this same thing, though it was a *little* more specific since it had access to the `RequestFailedException` thrown in this scenario, but it's a much more verbose change. Figured this was a nice, easier approach and good enough for when I need it.